### PR TITLE
Add CLI commands to deploy BigConfig files

### PR DESCRIPTION
**Pull request by scholtzan**
This PR adds a new CLI command group: `monitoring`. This group has 3 commands to interact with Bigeye:
- `bqetl monitoring validate <path>` - tries to load existing BigConfig files to see if their syntax is valid
- `bqetl monitoring update <path>` - creates a default BigConfig file if the `metadata.yaml` has `monitoring` enabled. Freshness and Volume get added as default metrics.
- `bqetl monitoring deploy <path>` - deploys BigConfig files to Bigeye

As follow up to this PR:
1. BigConfig files should be generated for the queries that have monitoring enabled via the CI so that BigConfig files are available as part of the `generated-sql` branch.
1. BigConfig files should be deployed in Airflow after the table deployments
1. We should update the docs

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [ ] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](https://github.com/mozilla/bigquery-etl/blob/main/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/DENG-4757)

### CODE REVIEW

File: sql/moz-fx-data-shared-prod/org_mozilla_fenix_derived/releases_v1/bigconfig.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+type: BIGCONFIG_FILE
+table_deployments:
+- deployments:
+  - fq_table_name: moz-fx-data-shared-prod.moz-fx-data-shared-prod.org_mozilla_fenix_derived.releases_v1
```
Comment (kik-kik) [inline sql/moz-fx-data-shared-prod/org_mozilla_fenix_derived/releases_v1/bigconfig.yml:4]: Is `moz-fx-data-shared-prod` being here double intentional?

File: bigquery_etl/cli/monitoring.py [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    invalid = False
+
+    for bigconfig_file in bigconfig_files:
+        try:
+            BigConfig.load(bigconfig_file)
+        except Exception as e:
+            click.echo(f"Invalid BigConfig file {bigconfig_file}: {e}")
+            invalid = True
+
+    if not invalid:
```
Comment (kik-kik) [inline bigquery_etl/cli/monitoring.py:177]: If we flip this around we could avoid the `else` here.

File: bigquery_etl/cli/monitoring.py [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+def monitoring(ctx):
+    """Create the CLI group for the monitoring command."""
+    pass
+
+
+@monitoring.command(
+    help="""
+    Deploy monitors defined in the BigConfig files to Bigeye.
+
+    Requires BigConfig credentials to be set via BIGEYE_API_CRED_FILE env variable.
```
Comment (kik-kik) [inline bigquery_etl/cli/monitoring.py:43]: Should we include the creation / set up of this inside `bootstrap.sh`?

File: bigquery_etl/cli/monitoring.py [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+    """Deploy monitors to Bigeye."""
+    metadata_files = paths_matching_name_pattern(
+        name, sql_dir, project_id=project_id, files=["metadata.yaml"]
+    )
+
+    for metadata_file in metadata_files:
+        project, dataset, table = extract_from_query_path(metadata_file)
+        try:
+            metadata = Metadata.from_file(metadata_file)
+            if metadata.monitoring and metadata.monitoring.enabled:
```
Comment (kik-kik) [inline bigquery_etl/cli/monitoring.py:60]: Is the `metadata.monitoring` needed here? If `metadata.monitoring.enabled` is `True` the previous has to be true anyways?

File: bigquery_etl/cli/monitoring.py [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+                ctx.invoke(update, name=name, sql_dir=sql_dir, project_id=project_id)
+                api_auth = ApiAuth.load_from_ini_file(
+                    auth_file=ApiAuth.find_user_credentials(None), workspace="DEFAULT"
```
Comment (kik-kik) [inline bigquery_etl/cli/monitoring.py:63]: Should `workspace` also be a parameter?

File: bigquery_etl/cli/monitoring.py [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+                )  # load user credentials from BIGEYE_API_CONFIG_FILE env variable
+                client = datawatch_client_factory(api_auth, workspace_id=463)
+                mc = MetricSuiteController(client=client)
+
+                mc.execute_bigconfig(
```
Comment (kik-kik) [inline bigquery_etl/cli/monitoring.py:68]: Should we have a call here to validate prior to deployment?

File: bigquery_etl/cli/monitoring.py [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+                    auto_approve=True,
+                )
+
+        except FileNotFoundError:
+            print("No metadata file for: {}.{}.{}".format(project, dataset, table))
+
+
+@monitoring.command(
+    help="""
+    Update BigConfig files based on monitoring metadata.
```
Comment (kik-kik) [inline bigquery_etl/cli/monitoring.py:83]: What is the difference here between deploy and update? Is there a need for us to have the distinction here?

File: sql/moz-fx-data-shared-prod/org_mozilla_fenix_derived/releases_v1/bigconfig.yml (hunk 1 above)
Comment (scholtzan) [reply sql/moz-fx-data-shared-prod/org_mozilla_fenix_derived/releases_v1/bigconfig.yml:4]: Yeah, so Bigeye is doing something really strange where the dataset name is actually `moz-fx-data-shared-prod.<dataset_name>`. I'll bring it up with them in a ticket

File: bigquery_etl/cli/monitoring.py (hunk 7 above)
Comment (scholtzan) [reply bigquery_etl/cli/monitoring.py:83]: Deploy will actually apply the BigConfig file to Bigeye.
Update is generating/updating the BigConfig file. Currently, only based on the `metadata.yaml` file.

File: bigquery_etl/cli/monitoring.py (hunk 4 above)
Comment (scholtzan) [reply bigquery_etl/cli/monitoring.py:60]: `metadata.monitoring` could be set to null or {}, then `.enabled` would raise an exception

**Pull request merged**